frontend/src/features/admin-form/settings/components/EndPageSettingsSection/EndPageSettingsInput.tsx

```diff
@@ -0,0 +1,93 @@
+import { useCallback } from 'react'
+import { useForm } from 'react-hook-form'
+import { Flex, FormControl, Stack } from '@chakra-ui/react'
+import validator from 'validator'
+
+import { FormEndPage } from '~shared/types'
+
+import Button from '~components/Button'
+import FormErrorMessage from '~components/FormControl/FormErrorMessage'
+import FormLabel from '~components/FormControl/FormLabel'
+import Input from '~components/Input'
+import Textarea from '~components/Textarea'
+
+import { useMutateFormPage } from '~features/admin-form/common/mutations'
+
+interface EndPageSettingsInputProps {
+  endPage: FormEndPage
+}
+
+export const EndPageSettingsInput = ({
+  endPage,
+}: EndPageSettingsInputProps): JSX.Element => {
+  const { mutateFormEndPage } = useMutateFormPage()
+
+  const buttonLinkValidation = useCallback(
+    (url) =>
+      !url ||
+      validator.isURL(url, {
+        protocols: ['https'],
+        require_protocol: true,
+      }) ||
+      'Please enter a valid URL (starting with https://)',
+    [],
+  )
```

Comment:
the whole validation object can actually be memoized; don't have to memoize just the function haha:

```tsx
const buttonLinkRules = useMemo(() => {
  return { 
    validate: (url) =>
      !url ||
      validator.isURL(url, {
        protocols: ['https'],
        require_protocol: true,
      }) ||
      'Please enter a valid URL (starting with https://)'
  }
}, [])
```


in fact, we can even instantiate it outside of this component as a constant.

Reply:
Alright thanks! Can't we only use `useMemo` within a React component?

Reply:
ya, if we instantiate outside the component don't need use `useMemo` la haha

Reply:
Okay thanks hahaha I'll instantiate outside